db/migrations/mysql/000012_create_preferences.up.sql

```diff
@@ -0,0 +1,105 @@
+CREATE TABLE IF NOT EXISTS Preferences (
+    UserId varchar(26) NOT NULL,
+    Category varchar(32) NOT NULL,
+    Name varchar(32) NOT NULL,
+    Value text,
+    PRIMARY KEY (UserId, Category, Name)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
+
+SET @preparedStatement = (SELECT IF(
+    (
+        SELECT COUNT(*) FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'Preferences'
+        AND table_schema = DATABASE()
+        AND column_name = 'Value'
+    ) > 0,
+    'ALTER TABLE Preferences MODIFY Value text;',
```

Comment:
What would happen if the value column already exists and is text? Should we not check on the type instead just on the existence?

Reply:
i think it will be an noop, but better to double check this in order to be sure.

Reply:
i've checked what store used to do, and we only care if the column exists: https://github.com/mattermost/mattermost-server/blob/master/store/sqlstore/store.go#L793

So i think we are achieving parity here, but i will spend some time to see if we can improve the check here in order to avoid cycles